libafl_bolts/src/tuples.rs

```diff
@@ -25,21 +25,33 @@ use crate::HasLen;
 use crate::Named;
 
 /// Returns if the type `T` is equal to `U`, ignoring lifetimes.
+#[inline] // this entire call gets optimized away :)
 #[must_use]
 pub fn type_eq<T: ?Sized, U: ?Sized>() -> bool {
+    // decider struct: hold a cell (which we will update if the types are unequal) and some
+    // phantom data using a function pointer to allow for Copy to be implemented
     struct W<'a, T: ?Sized, U: ?Sized>(&'a Cell<bool>, PhantomData<fn() -> (&'a T, &'a U)>);
 
+    // default implementation: if the types are unequal, we will use the clone implementation
     impl<'a, T: ?Sized, U: ?Sized> Clone for W<'a, T, U> {
         fn clone(&self) -> Self {
```

Comment:
This dude could also be `#[inline]` (it's in the same compile unit, so maybe it doesn't matter, but why not?)

Reply:
To be honest, inlining is unnecessary since the compiler will concretize this all away already. I'll add it for the debug build, I guess :grin: